### PR TITLE
docs: change "Configuring Proxy Concurrency" to reflect reality

### PR DESCRIPTION
**Pull request by hawkw**
Currently, the "Configuring Proxy Concurrency" Tasks documentation states that the proxy will always use the number of CPU cores available to the process. This is not actually correct --- if the `LINKERD2_PROXY_CORES` environment variable is unset, the proxy will always use a single-threaded runtime. If the environment variable is set, cgroup CPU limits will _also_ influence the number of worker threads spawned by the proxy, but when it is not present, the proxy will always run in single threaded mode.

Whether or not this is the ideal behavior is still open to discussion, but for now, the documentation should reflect reality. This change updates the docs to better describe the actual behavior.

Closes linkerd/linkerd2#10698

### CODE REVIEW

File: linkerd.io/content/2-edge/tasks/configuring-proxy-concurrency.md [hunk 1]
```diff
@@ -121,11 +121,7 @@ criteria must be met:
   This means that all containers in the pod must have both a limit and a request
   for memory and CPU, and the limit for each must have the same value as the
   request.
-- The CPU limit and CPU request must be an integer greater than or equal to 1.
-
-If you're not sure whether these criteria will all be met, it's best to use the
-`proxy-cpu-limit` annotation in addition to any Kubernetes CPU limits and
-requests.
+- The CPU limit and CPU request must be an integer greater than or
```
Comment (kflynn) [inline linkerd.io/content/2-edge/tasks/configuring-proxy-concurrency.md:124]: ...greater than or... what? 🙂

File: linkerd.io/content/2.13/tasks/configuring-proxy-concurrency.md [hunk 2]
```diff
@@ -121,11 +121,7 @@ criteria must be met:
   This means that all containers in the pod must have both a limit and a request
   for memory and CPU, and the limit for each must have the same value as the
   request.
-- The CPU limit and CPU request must be an integer greater than or equal to 1.
-
-If you're not sure whether these criteria will all be met, it's best to use the
-`proxy-cpu-limit` annotation in addition to any Kubernetes CPU limits and
-requests.
+- The CPU limit and CPU request must be an integer greater than or
```
Comment (kflynn) [inline linkerd.io/content/2.13/tasks/configuring-proxy-concurrency.md:124]: See previous comment. 🙂

**Review by kflynn — CHANGES_REQUESTED** [review]
Content looks great aside from a couple of unfinished sentences. 🙂 Should this be backported to 2.12?

File: linkerd.io/content/2-edge/tasks/configuring-proxy-concurrency.md (hunk 1 above)
Comment (hawkw) [reply linkerd.io/content/2-edge/tasks/configuring-proxy-concurrency.md:124]: whoops, looks like i accidentally deleted that
```suggestion
- The CPU limit and CPU request must be an integer greater than or equal to 1.
```

File: linkerd.io/content/2.13/tasks/configuring-proxy-concurrency.md (hunk 2 above)
Comment (hawkw) [reply linkerd.io/content/2.13/tasks/configuring-proxy-concurrency.md:124]:
```suggestion
- The CPU limit and CPU request must be an integer greater than or equal to 1.
```

File: linkerd.io/content/2-edge/tasks/configuring-proxy-concurrency.md [hunk 3]
```diff
@@ -94,23 +102,15 @@ CPU limits. This means that the Linux kernel is configured to limit the amount
 of time threads belonging to a given process are scheduled. Alternatively, the
 CPU manager policy may be set to
 [`static`](https://kubernetes.io/docs/tasks/administer-cluster/cpu-management-policies/#static-policy).
-In this case, the kubelet will use Linux `cgroup`s to enforce CPU limits for
+In this case, the kubelet use Linux `cgroup`s to enforce CPU limits for
```
[the remark this answers is not in the document]
Comment (hawkw) [reply linkerd.io/content/2-edge/tasks/configuring-proxy-concurrency.md:105]:
```suggestion
In this case, the kubelet will use Linux `cgroup`s to enforce CPU limits for
```

**Review by kflynn — APPROVED** [review]
Looks good from here, thanks!! 🙂

**Pull request merged**